nmea2000.orogen

```diff
@@ -9,48 +9,110 @@ using_task_library 'iodrivers_base'
 
 using_library 'nmea2000'
 import_types_from 'nmea2000/Message.hpp'
+import_types_from 'nmea2000/Receiver.hpp'
 import_types_from 'nmea2000Types.hpp'
 
 task_context 'CANTask' do
     needs_configuration
 
-    # The CAN messages
-    input_port 'can_in', '/canbus/Message'
+    # CAN message received by the bus
+    input_port('can_in', '/canbus/Message')
+        .needs_reliable_connection
 
-    # The NMEA2000 messages
-    output_port 'msg_out', '/nmea2000/Message'
+    # CAN messages sent to the bus
+    output_port 'can_out', '/canbus/Message'
+
+    # Whether we should continue enumerating devices even after the devices
+    # we do care about have been resolved
+    property 'continuous_device_enumeration', '/bool', false
+
+    # Expected devices with the name of the ports on which specific messages
+    # should be sent
+    property 'devices', '/std/vector</nmea2000/DeviceFilter>'
+
+    # How long the component should take before giving up on resolving
+    # devices listed in 'devices'. It transitions to DEVICE_RESOLUTION_FAILED
+    # when this timeout occurs
+    property 'resolution_timeout', '/base/Time'
+
+    # How long between the component queries product information when device
+    # resolution is needed
+    property 'enumeration_ack_timeout', '/base/Time'
+
+    # Dispatched messages as set up by the 'devices' property
+    dynamic_output_port(/\w+/, '/nmea2000/Message')
```

Comment:
Didn't know about `dynamic_output_port`... Is this new? What's the regex for?

Reply:
Not new. It's to tell Syskit that the component has the ability to create new ports. The regexp is to identify naming schemes when there are some (not the case here).

Reply:
- Syskit will refuse to generate networks with dynamic ports if there isn't a matching dynamic port declaration on the orogen file
- Syskit obviously verifies that the dynamic ports it expects have been created by the component

However, there is no help on the C++ side, that is the component implementer could very well create ports that do not match the declaration. I've been thinking for a while to add a `as: NAME` option to `dynamic_*_port` which would generate code to manage the ports for the component implementer, but never actually did it.